src/abstracts/SablierV2Lockup.sol

```diff
@@ -277,14 +281,29 @@ abstract contract SablierV2Lockup is
             revert Errors.SablierV2Lockup_Overdraw(streamId, amount, withdrawableAmount);
         }
 
+        // Retrieve the sender from storage.
+        address sender = getSender(streamId);
```

Comment:
> `getSender` before the internal `_withdraw` function is called for security reason, in case the sender is changed when the assets are transfered

Do we allow changing sender after stream has been created?

-------
Instead of loading sender from storage 2 times: 
1. `bool isCallerStreamSender = _isCallerStreamSender(streamId);`
2. `address sender = getSender(streamId);`

we can just load it once and re-use it from stack. It would save 100 gas.

Reply:
> Do we allow changing sender after stream has been created?

no, we don't, it was just a precaution in case some unwanted behaviour happens (better safe than sorry)

> we can just load it once and re-use it from stack. It would save 100 gas.

correct, but i suggest we leave it as it is, since, as you mentioned below, it is going to be removed anyway in your PR

Reply:
> it is just a precaution in case some unwanted behaviour happens

I understand your point and I think it's a good practice to take extra precautions. But I also think this is an overkill since there is no way to change sender after stream has been created. We don't have proxies so there is also no risk of overlapping storage slots. wdyt?

Nevertheless, in this case, we can leave it as it is because we, anyways, have to fetch `getSender(streamId)` after `_withdraw`.

> but i suggest we leave it as it is, since, as you mentioned below

OK. Agree.

Reply:
Am I correct that you ended undoing this behavior (i.e. the call to `getSender`)?

I agree with @smol-ninja that it's overkill to do this in production. The sender is unchangeable.

It might be worth adding a precautionary check like this in the tests, though.

Reply:
> Am I correct that you ended undoing this behavior (i.e. the call to getSender) in another PR

We no longer load sender from storage twice after #785 .

Reply:
got it